homeassistant/components/refoss/sensor.py

```diff
@@ -0,0 +1,176 @@
+        ),
+        RefossSensorEntityDescription(
+            key="energy",
+            name="This Month Energy",
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+            subkey="mConsume",
+            fn=lambda x: round(x, 2),
+        ),
+        RefossSensorEntityDescription(
+            key="energy_returned",
+            name="This Month Energy Returned",
+            device_class=SensorDeviceClass.ENERGY,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+            subkey="mConsume",
+            fn=lambda x: round(x, 2),
+        ),
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Refoss device from a config entry."""
+
+    @callback
+    def init_device(coordinator):
+        """Register the device."""
+        device = coordinator.device
+
+        if not isinstance(device, ElectricityXMix):
+            return
+        descriptions = SENSORS.get(device.device_type)
+        new_entities = []
+        for channel in device.channels:
+            for description in descriptions:
+                entity = RefossSensor(
+                    coordinator=coordinator,
+                    channel=channel,
+                    description=description,
+                )
+                new_entities.append(entity)
+
+        async_add_entities(new_entities)
+
+    for coordinator in hass.data[DOMAIN][COORDINATORS]:
+        init_device(coordinator)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, DISPATCH_DEVICE_DISCOVERED, init_device)
+    )
+
+
+class RefossSensor(RefossEntity, SensorEntity):
+    """Refoss Sensor Device."""
+
+    entity_description: RefossSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: RefossDataUpdateCoordinator,
+        channel: int,
+        description: RefossSensorEntityDescription,
+    ) -> None:
+        """Init Refoss sensor."""
+        super().__init__(coordinator, channel)
+        self.entity_description = description
+        self._attr_unique_id = f"{super().unique_id}{description.key}"
+        device_type = coordinator.device.device_type
+        self._attr_name = (
+            f"{CHANNEL_DISPLAY_NAME[device_type][channel]} {description.name}"
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the native value."""
+        value = self.coordinator.device.get_value(
+            self.channel_id, self.entity_description.subkey
+        )
+        if value is None:
+            return None
+        if (self.entity_description.key == ENERGY and value < 0) or (
+            self.entity_description.key == ENERGY_RETURNED and value > 0
+        ):
+            value = 0
```

Comment:
This should be moved inside of the value function `fn` of the entity description as it applies only to two specific entities